src/ui/components/tenants/tenantDetail/LoginMethodsSection.tsx

```diff
@@ -159,29 +157,33 @@ const LoginFactor = ({
 					} else {
 						throw new Error(res.status);
 					}
+				} else {
+					setError(null);
 				}
 			} else {
 				const res = await updateSecondaryFactors(tenantInfo.tenantId, id, !doesFactorExist);
 				if (res.status !== "OK") {
 					if (res.status === "RECIPE_NOT_CONFIGURED_ON_BACKEND_SDK_ERROR") {
 						setError(res.message);
 					} else if (res.status === "MFA_NOT_INITIALIZED_ERROR") {
-						setMfaError("MFA_NOT_INITIALIZED");
+						setMfaError?.("MFA_NOT_INITIALIZED");
 					} else {
 						throw new Error(res.status);
 					}
+				} else {
+					setError(null);
```

Comment:
The logic comes here when the status is OK. This is there to reset the error in case there was an error with the factor previously but now when toggling again the error is no longer there so we reset the error. We only show error for the factor if error is non null